src/script/conversation/MessageRepository.ts

```diff
@@ -603,23 +615,47 @@ export class MessageRepository {
    * @param conversation The conversation to send the message in
    * @returns Resolves to true if the message can be sent, false if the user didn't give their permission
    */
-  requestUserSendingPermission(conversation: Conversation, showLegalHoldWarning: boolean): Promise<boolean> {
+  async requestUserSendingPermission(
+    conversation: Conversation,
+    showLegalHoldWarning: boolean,
+    consentType: CONSENT_TYPE,
+  ): Promise<boolean> {
     const conversationDegraded = conversation.verification_state() === ConversationVerificationState.DEGRADED;
     if (showLegalHoldWarning) {
       return showLegalHoldWarningModal(conversation, conversationDegraded)
         .then(() => true)
         .catch(() => false);
     }
     if (!conversationDegraded) {
-      return Promise.resolve(true);
+      return true;
     }
 
     const users = conversation.getUsersWithUnverifiedClients();
     const userNames = joinNames(users, Declension.NOMINATIVE);
     const titleSubstitutions = capitalizeFirstChar(userNames);
 
-    const actionString = t('modalConversationNewDeviceAction');
-    const messageString = t('modalConversationNewDeviceMessage');
+    let actionString = t('modalConversationNewDeviceAction');
+    let messageString = t('modalConversationNewDeviceMessage');
+    switch (consentType) {
+      case CONSENT_TYPE.INCOMING_CALL: {
+        actionString = t('modalConversationNewDeviceIncomingCallAction');
+        messageString = t('modalConversationNewDeviceIncomingCallMessage');
+        break;
+      }
+
+      case CONSENT_TYPE.OUTGOING_CALL: {
+        actionString = t('modalConversationNewDeviceOutgoingCallAction');
+        messageString = t('modalConversationNewDeviceOutgoingCallMessage');
+        break;
+      }
+
+      default: {
```

Comment:
The default feels redundant, as these values are already in those vars at this point.